Sources/PackagePlugin/Context.swift

```diff
@@ -64,12 +64,9 @@ public struct PluginContext {
         throw PluginContextError.toolNotFound(name: name)
     }
 
-    /// A mapping from tool names to their definitions. Not directly available
+    /// A mapping from tool names to their paths and triples. Not directly available
     /// to the plugin, but used by the `tool(named:)` API.
-    let toolNamesToPaths: [String: Path]
-
-    /// Supported triples per tool name; looked up in `tool(named:)`
-    let toolNamesToTriples: [String: [String]]
+    let accessibleTools: [String: (path: Path, triples: [String]?)]
```

Comment:
Is there any semantic difference to having `.none` for triples vs `[]`?  In other words, should this really be optional or would just a plain array suffice?

Reply:
Hm, I think I see:  `.none` means "any" (for example if the tool is actually a shell script that runs anywhere) and `[]` means "none" (i.e. that there are no compatible triples with the host).

Reply:
Yep, that's right. Basically for anything built from source, triples is `nil`, but for vended tools, there's always a value and `[]` means "no compatible triples".